src/Simplify.cpp

```diff
@@ -1465,6 +1490,56 @@ class Simplify : public IRMutator {
             // (y - x*4) / 2 -> y/2 - x*2
             Expr ratio = make_const(op->type, div_imp(ia, ib));
             expr = mutate((sub_a->a / b) - (mul_a_b->a * ratio));
+        } else if (no_overflow(op->type) && add_a && add_a_a &&
+                   mul_a_a_a && const_int(mul_a_a_a->b, &ia) && const_int(b, &ib) &&
+                   ib > 0 && (ia % ib == 0)) {
+            // Pull terms that are a multiple of the divisor out
+            // ((x*4 + y) + z) / 2 -> x*2 + (y + z)/2
+            Expr ratio = make_const(op->type, div_imp(ia, ib));
+            expr = mutate((mul_a_a_a->a * ratio) + (add_a_a->b  + add_a->b) / b);
+        } else if (no_overflow(op->type) && add_a && sub_a_a &&
+                   mul_a_a_a && const_int(mul_a_a_a->b, &ia) && const_int(b, &ib) &&
+                   ib > 0 && (ia % ib == 0)) {
+            // ((x*4 - y) + z) / 2 -> x*2 + (z - y)/2
+            Expr ratio = make_const(op->type, div_imp(ia, ib));
+            expr = mutate((mul_a_a_a->a * ratio) + (add_a->b - sub_a_a->b) / b);
+        } else if (no_overflow(op->type) && sub_a && add_a_a &&
+                   mul_a_a_a && const_int(mul_a_a_a->b, &ia) && const_int(b, &ib) &&
+                   ib > 0 && (ia % ib == 0)) {
+            // ((x*4 + y) - z) / 2 -> x*2 + (y - z)/2
+            Expr ratio = make_const(op->type, div_imp(ia, ib));
+            expr = mutate((mul_a_a_a->a * ratio) + (add_a_a->b - sub_a->b) / b);
+        } else if (no_overflow(op->type) && sub_a && sub_a_a &&
```

Comment:
Please put every clause on its own line, like the other rules.

Generally we just wrap somewhere near 80 characters, but in the simplifier the clauses get long and difficult to read, so we use one line per and clause instead.